### PR TITLE
TYP and DOC: pd.merge accepts Series

### PR DESCRIPTION
**Pull request by Dr-Irv**
I noticed that the docs for `pd.merge()` and `pd.merge_asof()` were requiring the first argument to be a `DataFrame`, but in fact, they can be a `Series`.  This PR updates the docs and the types to fix that issue.

Didn't think a `whatsnew` entry was needed, and I skipped creating an issue, but will do so if required.

### CODE REVIEW

**Review by lithomas1 — COMMENTED** [review]
I think you should use DataFrame | Series instead of FrameOrSeriesUnion. (See #40876 for more context)

**Comment by Dr-Irv** [comment]
> I think you should use DataFrame | Series instead of FrameOrSeriesUnion. (See #40876 for more context)

Thanks for pointing that out.  Have now changed it.

**Review by jreback — APPROVED** [review]
lgtm

**Comment by jreback** [comment]
cc @simonjayhawkins if any comments.

File: pandas/core/reshape/merge.py [hunk 1]
```diff
@@ -362,8 +362,8 @@ def merge_asof(
 
     Parameters
     ----------
-    left : DataFrame
-    right : DataFrame
+    left : DataFrame or named Series
+    right : DataFrame or named Series
```
Comment (simonjayhawkins) [inline pandas/core/reshape/merge.py:368]: DataFrame is mentioned quite often elsewhere in the docstring. Maybe should do a pass on the complete docstring for clarity.

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @Dr-Irv

File: pandas/core/reshape/merge.py (hunk 1 above)
Comment (Dr-Irv) [reply pandas/core/reshape/merge.py:368]: I started to change all references from "DataFrame" or "frame" to say "DataFrame or Series", and it became quite clumsy.  So I added a line at the top of the docs to say that a `Series` is treated as a `DataFrame` with a single named column, and then the rest of the docs works.

**Comment by simonjayhawkins** [comment]
Thanks @Dr-Irv

**Pull request merged**